compiler/src/main/java/com/squareup/anvil/compiler/codegen/reference/AnnotationReferenceIr.kt

```diff
@@ -44,35 +44,39 @@ internal class AnnotationReferenceIr(
     )
 
   val parentScope: ClassReferenceIr by lazy(NONE) {
-    context.referenceClass(annotation.parentScope())?.toClassReference(context)
+    context.referenceClass(argumentFqName("parentScope"))?.toClassReference(context)
       ?: throw AnvilCompilationException(
         element = annotation,
         message = "Couldn't find parent scope for $fqName."
       )
   }
 
   val excludedClasses: List<ClassReferenceIr> by lazy(NONE) {
-    argumentClassArray("exclude")
+    argument("exclude")?.value<List<ClassReferenceIr>>().orEmpty()
   }
 
   val replacedClasses: List<ClassReferenceIr> by lazy(NONE) {
-    argumentClassArray("replaces")
+    argument("replaces")?.value<List<ClassReferenceIr>>().orEmpty()
   }
 
-  // TODO: May be worth introducing AnnotationArgumentIr here
-  fun argumentClassArray(name: String): List<ClassReferenceIr> {
-    return annotation.argumentClassArray(name)
-  }
+  private fun argumentFqName(name: String): FqName {
```

Comment:
I'm a little confused why this function is needed. It should be a call similar to `excludedClasses`:
```kotlin
  val scopeOrNull: ClassReferenceIr? by lazy(NONE) {
    argument("scope")?.value<ClassReferenceIr>()
  }
```

Reply:
Good catch, I didn't connect that this could just be replaced now instead of moved+refactored from the utils.